### PR TITLE
02 simple golden acorn app

### PR DESCRIPTION
**Pull request by evasimonyi**
please review my code

previously working app codebase is refactored:
components are in separate files
prop types are added
app is working

### CODE REVIEW

File: simple-golden-acorn-app-with-states/src/components/button.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"use strict";
```
Comment (gyorkibence) [inline simple-golden-acorn-app-with-states/src/components/button.js]: u dont have to use usestrict at react files, pls delete them

File: simple-golden-acorn-app-with-states/src/components/button.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import React from "react";
+import PropTypes from "prop-types";
+
+const Button = ({ name, clickEvent }) => {
+  return (
```
Comment (gyorkibence) [inline simple-golden-acorn-app-with-states/src/components/button.js]: delete the return, just simply use => ( jsx codes );

File: simple-golden-acorn-app-with-states/src/components/button.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <button name={name} onClick={clickEvent}>
+      {name}
+    </button>
+  );
+};
+
+Button.propTypes = {
+  count: PropTypes.func.isRequired
```
Comment (gyorkibence) [inline simple-golden-acorn-app-with-states/src/components/button.js]: why count? why function?, name and clickEvent :D

File: simple-golden-acorn-app-with-states/src/components/button.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+"use strict";
+
+import React from "react";
+import PropTypes from "prop-types";
+
+const Button = ({ name, clickEvent }) => {
+  return (
+    <button name={name} onClick={clickEvent}>
```
Comment (gyorkibence) [inline simple-golden-acorn-app-with-states/src/components/button.js]: u dont need the name attribute on this button tag

File: simple-golden-acorn-app-with-states/src/components/counter.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    if (this.state.count > 0) {
+      this.setState({ count: this.state.count - 1 });
+    }
+  };
+
+  render() {
+    return (
+      <div>
+        <Button clickEvent={this.handleIncrement} name="get one 🌰 " />
+        <span>
```
Comment (gyorkibence) [inline simple-golden-acorn-app-with-states/src/components/counter.js]: whats the purpose of this span?

File: simple-golden-acorn-app-with-states/src/components/button.js (hunk 3 above)
Comment (evasimonyi) [reply simple-golden-acorn-app-with-states/src/components/button.js]: corrected

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickEvent: PropTypes.func.isRequired
};

File: simple-golden-acorn-app-with-states/src/components/button.js (hunk 4 above)
Comment (evasimonyi) [reply simple-golden-acorn-app-with-states/src/components/button.js]: so like this?
const Button = ({ name, clickEvent }) => {
  <button onClick={clickEvent}>{name}</button>;
};

File: simple-golden-acorn-app-with-states/src/components/counter.js (hunk 5 above)
Comment (evasimonyi) [reply simple-golden-acorn-app-with-states/src/components/counter.js]: I accidentally left it there

**Pull request merged**